### PR TITLE
Support labelallow stage in Promtail

### PR DESCRIPTION
**Pull request by adityacs**
**What this PR does / why we need it**:
Support `labelallow` stage in Promtail

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/clients/promtail/stages/labelallow.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - kubernetes_pod_name
+    - kubernetes_container_name
+```
+
+Given the following incoming labels:
+
+- `kubernetes_pod_name`: `"loki-pqrs"`
+- `kubernetes_container_name`: `"loki"`
+- `kubernetes_pod_template_hash`: `"79f5db67b"`
+- `kubernetes_controller_revision_hash`: `"774858987d"`
```
Comment (adityacs) [inline docs/sources/clients/promtail/stages/labelallow.md:36]: Do we need a better example?

**Review by slim-bean — CHANGES_REQUESTED** [review]
Thanks @adityacs this is great! could you also update the doc page which lists all the stages the _index.md file in the stages folder?

**Comment by adityacs** [comment]
@slim-bean Updated `_index.md`

**Review by slim-bean — APPROVED** [review]
LGTM!

**Pull request merged**